edb/edgeql/compiler/__init__.py

```diff
@@ -144,7 +145,8 @@ def compile_ast_to_ir(tree,
         implicit_tid_in_shapes=implicit_tid_in_shapes,
         schema_view_mode=schema_view_mode,
         disable_constant_folding=disable_constant_folding,
-        json_parameters=json_parameters)
+        json_parameters=json_parameters,
+        session_mode=session_mode)
```

Comment:
Session mode is a global switch, so it should be in the `Environment` portion of the context.

Reply:
OK, refactored to be an `Environment` attribute.